File: src/config/overrides.ts
import { DoorClosed, Lock, ShieldCheck } from "lucide-react";
import type { OverrideMap } from "../lib/resolve";

/**
 * Project-local label/icon overrides. Seeded from the owner's stock-HA "before"
 * screenshot, where the single Schlage and its Z-Wave diagnostic sensors render
 * with raw, attribute-derived names ("Lock", "Lock Current status …", "Lock
 * Intrusion"). These force the human labels Hawksnest is built to guarantee.
 *
 * In Phase 1 this map remains the highest-priority tier over the live registry.
 */
export const overrides: OverrideMap = {
  "lock.front_door_lock": { name: "Front Door", icon: Lock },
  // HA exposes this contact sensor as "Lock Current status …" — the poster child.
  "binary_sensor.front_door_current_status": {
    name: "Front Door",
    icon: DoorClosed,
  },
  "binary_sensor.front_door_intrusion": {
    name: "Intrusion",
    icon: ShieldCheck,
  },
  "lock.back_door_lock": { name: "Back Door", icon: Lock },
  "light.basement": { name: "Basement Lights" },
  "camera.front_door": { name: "Front Door" },
  "alarm_control_panel.home": { name: "Home Alarm" },
};

File: android/app/src/main/java/com/hawksnest/config/Overrides.kt
package com.hawksnest.config

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.DoorFront
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Shield
import com.hawksnest.core.logic.EntityOverride
import com.hawksnest.core.logic.OverrideMap

/**
 * Project-local label/icon overrides, seeded from the owner's stock-HA "before" screenshot where
 * the Schlage lock and its Z-Wave diagnostic sensors render with raw names ("Lock Current status
 * …"). These force the human labels Hawksnest guarantees. Highest-priority tier over the live
 * registry. Ported from `src/config/overrides.ts`.
 */
val overrides: OverrideMap = mapOf(
    "lock.front_door_lock" to EntityOverride(name = "Front Door", icon = Icons.Filled.Lock),
    "binary_sensor.front_door_current_status" to
        EntityOverride(name = "Front Door", icon = Icons.Filled.DoorFront),
    "binary_sensor.front_door_intrusion" to
        EntityOverride(name = "Intrusion", icon = Icons.Filled.Shield),
    "lock.back_door_lock" to EntityOverride(name = "Back Door", icon = Icons.Filled.Lock),
    "light.basement" to EntityOverride(name = "Basement Lights"),
    "camera.front_door" to EntityOverride(name = "Front Door"),
    "alarm_control_panel.home" to EntityOverride(name = "Home Alarm"),
)
